src/adapter/src/catalog.rs

```diff
@@ -1082,6 +1096,7 @@ impl CatalogItem {
             CatalogItem::Func(_) => mz_sql::catalog::CatalogItemType::Func,
             CatalogItem::Secret(_) => mz_sql::catalog::CatalogItemType::Secret,
             CatalogItem::Connection(_) => mz_sql::catalog::CatalogItemType::Connection,
+            CatalogItem::StorageCollection(_) => mz_sql::catalog::CatalogItemType::Source,
```

Comment:
Good question! I'm just reusing it because the logs also reuse it, but I'm not entirely sure of the implications. From what I could tell from looking at the usages of `CatalogItemType`, all the `Source` behaviors are appropriate, but I might be missing something.

Reply:
Yeah, I think thinking of this as a "built-in source" makes sense. There may yet be some safety issues we're not considering, but surely less than there would be if we called this a "table" or an "index"! And I like not introducing another noun that users need to deal with; this should present as a source in all user-facing contexts.